symfony/plugins/orangehrmAdminPlugin/Api/LocationAPI.php

```diff
@@ -0,0 +1,391 @@
+
+    /**
+     * @return LocationService
+     */
+    public function getLocationService(): LocationService
+    {
+        if (is_null($this->locationService)) {
+            $this->locationService = new LocationService();
+        }
+        return $this->locationService;
+    }
+
+    /**
+     * @param LocationService $locationService
+     */
+    public function setLocationService(LocationService $locationService): void
+    {
+        $this->locationService = $locationService;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetOne(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(CommonParams::PARAMETER_ID, new Rule(Rules::POSITIVE)),
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getOne(): EndpointResourceResult
+    {
+        $id = $this->getRequestParams()->getInt(RequestParams::PARAM_TYPE_ATTRIBUTE, CommonParams::PARAMETER_ID);
+        $location = $this->getLocationService()->getLocationById($id);
+        $this->throwRecordNotFoundExceptionIfNotExist($location, Location::class);
+        return new EndpointResourceResult(LocationModel::class, $location);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getAll(): EndpointResult
+    {
+        $locationSearchFilterParams = new LocationSearchFilterParams();
+
+        $this->setSortingAndPaginationParams($locationSearchFilterParams);
+
+        $locationSearchFilterParams->setName(
+            $this->getRequestParams()->getStringOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_LOCATION_NAME
+            )
+        );
+        $locationSearchFilterParams->setCity(
+            $this->getRequestParams()->getStringOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_LOCATION_CITY_NAME
+            )
+        );
+        $locationSearchFilterParams->setCountryCode(
+            $this->getRequestParams()->getStringOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_LOCATION_COUNTRY_CODE
+            )
+        );
+
+        $locations = $this->getLocationService()->searchLocations($locationSearchFilterParams);
+
+        return new EndpointCollectionResult(
+            LocationModel::class,
+            $locations,
+            new ParameterBag(
+                [
+                    CommonParams::PARAMETER_TOTAL => $this->getLocationService()->getSearchLocationListCount(
+                        $locationSearchFilterParams
+                    ),
+                ]
+            )
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(self::FILTER_LOCATION_NAME),
+            new ParamRule(self::FILTER_LOCATION_CITY_NAME),
+            new ParamRule(self::FILTER_LOCATION_COUNTRY_CODE),
```

Comment:
Can you refer https://github.com/orangehrm/orangehrm/blob/5.0/symfony/plugins/orangehrmPimPlugin/Api/EmployeeContactDetailsAPI.php#L191-L196

May be not matching to your requirement, have a look

Reply:
Since this validation method is for getAll (for validating search filters), do we need to validate them with lengths and other attributes?